eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/comm/requests/m2w/M2WebEwonUpdateTagValuesRequest.java

```diff
@@ -0,0 +1,118 @@
+
+  /**
+   * Constructs a new {@link M2WebEwonUpdateTagValuesRequest} object with the specified M2Web/Talk2M
+   * server session ID, {@link CommunicationAuthInfo}, and Ewon name.
+   *
+   * @param server The M2Web/Talk2M server to perform the Ewon update tag values request on.
+   * @param t2mSessionId The Talk2M session ID to authenticate with.
+   * @param communicationAuthInfo The Talk2M account {@link CommunicationAuthInfo} object to use for
+   *     the request.
+   * @param ewonName The name of the Ewon to perform the update tag values request on.
+   * @param tagValuePairs The tag name and (updated) value pairs.
+   * @since 1.0.0
+   */
+  public M2WebEwonUpdateTagValuesRequest(
+      String server,
+      String t2mSessionId,
+      CommunicationAuthInfo communicationAuthInfo,
+      String ewonName,
+      KeyValue<String, String>[] tagValuePairs) {
+    this(
+        server,
+        t2mSessionId,
+        communicationAuthInfo.getDevId(),
+        communicationAuthInfo.getEwonUsername(),
+        communicationAuthInfo.getEwonPassword(),
+        ewonName,
+        tagValuePairs);
+  }
+
+  /**
+   * Constructs a new {@link M2WebEwonUpdateTagValuesRequest} object with the specified M2Web/Talk2M
+   * server, session ID, developer ID, device username, device password, and Ewon name.
+   *
+   * @param server The M2Web/Talk2M server to perform the Ewon update tag values request on.
+   * @param t2mSessionId The Talk2M session ID to authenticate with.
+   * @param t2mDeveloperId The Talk2M developer ID to use for the request.
+   * @param t2mDeviceUsername The Talk2M device username to use for the request. Can not be null.
+   * @param t2mDevicePassword The Talk2M device password to use for the request. Can not be null.
+   * @param ewonName The name of the Ewon to perform the update tag values request on.
+   * @param tagValuePairs The tag name and (updated) value pairs.
+   * @since 1.0.0
+   */
+  public M2WebEwonUpdateTagValuesRequest(
+      String server,
+      String t2mSessionId,
+      String t2mDeveloperId,
+      String t2mDeviceUsername,
+      String t2mDevicePassword,
+      String ewonName,
+      KeyValue<String, String>[] tagValuePairs) {
+    super(
+        server,
+        t2mSessionId,
+        t2mDeveloperId,
+        t2mDeviceUsername,
+        t2mDevicePassword,
+        ewonName,
+        getUpdateTagValuesRequestPath(tagValuePairs));
+  }
+
+  /**
+   * Gets the Ewon update tag values request path for the specified tag name and (updated) value
+   * pairs.
+   *
+   * @param tagValuePairs The tag name and (updated) value pairs.
+   * @return The Ewon update tag values request path.
+   * @since 1.0.0
+   */
+  private static String getUpdateTagValuesRequestPath(KeyValue<String, String>[] tagValuePairs) {
+    StringBuilder requestPath = new StringBuilder(UPDATE_TAG_VALUES_GET_REQUEST_PATH_BASE);
+
+    // Add tag name and value pairs to request path
+    for (int i = 0; i < tagValuePairs.length; i++) {
```

Comment:
why is i not something meaningful like tagValuePairIndex?

Reply:
Updated